docs/adr/0039-locales-languages.md

```diff
@@ -0,0 +1,183 @@
+
+```
+> locale = new Intl.Locale("fr-fr")
+> locale.toString()
+'fr-FR'
+> locale.language
+'fr'
+> locale.region
+'FR'
+```
+
+#### Monnaie (currency)
+
+Alors que par définition les paramètres régionaux contiennent la monnaie, le builtin [`Intl.Locale`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale) ne contient rien sur le système de monnaie (currency) d'une région. A priori ce n'est pas un besoin pour Pix, mais on le mentionne ici car il faudrait creuser pour comprendre le pourquoi de cette exception, pour ne pas risquer de manquer des notions/détails potentiellement importants.
+
+### État du code de Pix
+
+Actuellement dans le code de pix les 2 notions et noms de *locale* et *language* sont manipulés :
+
+* la notion et le nom de *language* sont utilisés pour le côté client
+* la notion et le nom de *locale* sont utilisés côté BDD et hapi
+
+En ce qui concerne le format des identifiants de *language*/*locale*, c'est le format suivant qui est utilisé dans le code de pix : `fr`, `fr-fr`, `fr-be`. Ce format n'est pas la forme canonique de représentation tel que défini par la spécification *BCP 47* (car la partie `region` devrait être en majuscules), mais c'est une forme acceptée et par spécification c'est compatible avec `Intl.Locale`.
+
+
+## Solution proposée : Fonder tous les traitements sur la notion de locales
+
+**Description**
+
+Dans le cas de Pix c'est la notion de *locale* qui est la plus importante car une *locale* est une association langue + autres données liées au groupe auquel on s'adresse, ce qui correspond bien aux cas d'utilisation de Pix où chaque utilisateur fait partie d'un groupe linguistique-géolocalisé, même si la géolocalisation de ce groupe peut parfois être très large.
+
+On souhaite donc :
+
+* associer à chaque utilisateur une *locale* (et non plus un *language*) au format BCP 47 avec persistance de ce choix en base de données
+* avoir les différentes versions des sites web identifiées par une *locale* au format BCP 47 (`fr`, `fr-FR`, `fr-BE`, `en`) et non plus des identifiants non-canoniques (`fr`, `fr-fr`, `fr-be`, `en` ont la partie `region` en minuscules)
+* Utiliser le builtin JavaScript [`Intl.Locale`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale) chaque fois que c'est possible
+
+**Avantage(s) :**
+
+- Traiter systématiquement des *locales* est une recommandation facile à comprendre et à suivre
+- Le builtin JavaScript `Intl.Locale` est un outil disponible. Même si `Intl.Locale` n'est pas disponible sur tous les navigateurs on pourrait ne l'utiliser d'abord que côté serveur et utiliser sa forme sérialisée par la méthode `toString` partout ailleurs.
+
+**Inconvénient(s) :**
+
+- Actuellement la table `users` a une colonne `lang` et non `locale`, ce qui demanderait du travail de migration de données et de code si on veut passer à `locale`
```

Comment:
Autre "inconvénient" ou point à creuser : le passage des sites vitrines `fr` et `fr-BE` qui arrive sur l'app pour n'avoir que du `fr`, et la correspondance entre ces différentes locales.

Reply:
Pour les locales il y a la notion de match, qui cherche la meilleure correspondance possible. Par exemple si un utilisateur demandant du `fr-BE` arrive sur un site web/une application ne disposant que du `fr` et du `en`, alors l'application/site web doit proposer alors à cet utilisateur `fr`. J'espère que j'ai bien compris ton point. Je vais ajouter un point sur le match/meilleure correspondance possible des locales dans le document. Merci !